src/messages/embed/Embed.ts

```diff
@@ -0,0 +1,304 @@
+			(this.description?.length ?? 0) +
+			(this.fields.length >= 1
+				? this.fields.reduce((prev, curr) => prev + curr.name.length + curr.value.length, 0)
+				: 0) +
+			(this.footer?.text.length ?? 0) +
+			(this.author?.name?.length ?? 0)
+		);
+	}
+
+	/**
+	 * Adds a field to the embed (max 25).
+	 * @param name The name of this field.
+	 * @param value The value of this field.
+	 * @param inline If this field will be displayed inline.
+	 */
+	public addField(name: string, value: string, inline = false): this {
+		return this.addFields({ name, value, inline });
+	}
+
+	/**
+	 * Adds fields to the embed (max 25).
+	 * @param fields The fields to add.
+	 */
+	public addFields(...fields: APIEmbedField[]): this {
+		// Data assertions
+		ow(fields, embedFieldsArrayPredicate);
+
+		// Ensure adding this field won't exceed the 25 field limit
+		validateFieldLength(this.fields, fields.length);
+
+		this.fields.push(...Embed.normalizeFields(...fields));
+		return this;
+	}
+
+	/**
+	 * Removes, replaces, and inserts fields in the embed (max 25).
+	 * @param index The index to start at.
+	 * @param deleteCount The number of fields to remove.
+	 * @param fields The replacing field objects.
+	 */
+	public spliceFields(index: number, deleteCount: number, ...fields: APIEmbedField[]): this {
+		// Data assertions
+		ow(fields, 'fields', embedFieldsArrayPredicate);
+
+		// Ensure adding this field won't exceed the 25 field limit
+		validateFieldLength(this.fields, fields.length - deleteCount);
+
+		this.fields.splice(index, deleteCount, ...Embed.normalizeFields(...fields));
+		return this;
+	}
+
+	/**
+	 * Sets the author of this embed.
+	 * @param name The name of the author.
+	 * @param options The URL options of the author.
+	 * @param options.iconURL The icon URL of the author.
+	 * @param options.url The URL of the author.
+	 */
+	public setAuthor(name: string | null, { iconURL, url }: AuthorURLOptions = {}): this {
+		// Data assertions
+		ow(name, 'name', authorNamePredicate);
+		ow(iconURL, 'iconURL', urlPredicate);
+		ow(url, 'url', urlPredicate);
+
+		this.author = name ? { name, icon_url: iconURL ?? undefined, url: url ?? undefined } : undefined;
+		return this;
+	}
+
+	/**
+	 * Sets the color of this embed.
+	 * @param color The color of the embed.
+	 */
+	public setColor(color: number | null): this {
+		// Data assertions
+		ow(color, 'color', colorPredicate);
+
+		this.color = color ?? undefined;
+		return this;
+	}
+
+	/**
+	 * Sets the description of this embed.
+	 * @param description The description.
+	 */
+	public setDescription(description: string | null): this {
+		// Data assertions
+		ow(description, 'description', descriptionPredicate);
+
+		this.description = description ?? undefined;
+		return this;
+	}
+
+	/**
+	 * Sets the footer of this embed.
+	 * @param text The text of the footer.
+	 * @param iconURL The icon URL of the footer.
+	 */
+	public setFooter(text: string, iconURL?: string | null): this {
```

Comment:
Sorts same as with setAuthor, tho I don't think having a function for every option is feasible (think setFooterText, setFooterIconURL, etc). @discordjs/the-big-3 here too